tests/chainer_tests/functions_tests/math_tests/test_unary.py

```diff
@@ -0,0 +1,315 @@
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_det_scaling_gpu(self):
+        self.det_scaling(gpu=True)
+
+    @condition.retry(3)
+    def test_det_scaling_cpu(self):
+        self.det_scaling(gpu=False)
+
+    def det_identity(self, gpu=False):
+        idt = [numpy.identity(self.x.shape[1]).astype(numpy.float32)
+               for _ in range(self.x.shape[0])]
+        idt = numpy.array(idt).astype(numpy.float32)
+        chk = numpy.ones(self.x.shape[0]).astype(numpy.float32)
+        if gpu:
+            chk = cuda.to_gpu(chk)
+            idt = cuda.to_gpu(idt)
+        idtv = chainer.Variable(idt)
+        idtd = F.batch_det(idtv)
+        gradient_check.assert_allclose(idtd.data, chk, rtol=1e-4, atol=1e-4)
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_det_identity_gpu(self):
+        self.det_identity(gpu=True)
+
+    @condition.retry(3)
+    def test_det_identity_cpu(self):
+        self.det_identity(gpu=False)
+
+    def det_product(self, gpu=False):
+        if gpu:
+            cx = cuda.to_gpu(self.x.copy())
+            cy = cuda.to_gpu(self.y.copy())
+        else:
+            cx = self.x.copy()
+            cy = self.y.copy()
+        vx = chainer.Variable(cx)
+        vy = chainer.Variable(cy)
+        dxy1 = F.batch_det(F.batch_matmul(vx, vy))
+        dxy2 = F.batch_det(vx) * F.batch_det(vy)
+        gradient_check.assert_allclose(dxy1.data, dxy2.data, rtol=1e-4,
+                                       atol=1e-4)
+
+    @condition.retry(3)
+    def test_det_product_cpu(self):
+        self.det_product(gpu=False)
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_det_product_gpu(self):
+        self.det_product(gpu=True)
+
+    def check_backward(self, x_data, y_grad, op, **kwargs):
+        x = chainer.Variable(x_data)
+        y = op(x)
+        y.grad = y_grad
+        y.backward()
+
+        func = y.creator
+        f = lambda: func.forward((x.data,))
+        gx, = gradient_check.numerical_grad(f, (x.data,), (y.grad,))
+
+        gradient_check.assert_allclose(gx, x.grad, **kwargs)
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_batch_backward_gpu(self):
+        x_data = cuda.to_gpu(self.x.copy())
+        y_grad = cuda.to_gpu(self.gy.copy())
+        self.check_backward(x_data, y_grad, F.batch_det)
+
+    @condition.retry(3)
+    def test_batch_backward_cpu(self):
+        x_data, y_grad = self.x.copy(), self.gy.copy()
+        self.check_backward(x_data, y_grad, F.batch_det)
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_backward_gpu(self):
+        x_data = cuda.to_gpu(self.x[0].copy())
+        y_grad = cuda.to_gpu(self.gy[0][None].copy())
+        self.check_backward(x_data, y_grad, F.det)
+
+    @condition.retry(3)
+    def test_backward_cpu(self):
+        x_data, y_grad = self.x[0].copy(), self.gy[0][None].copy()
+        self.check_backward(x_data, y_grad, F.det)
+
+
+class TestUnaryFunctionsSimple(UnaryFunctionsTestBase, unittest.TestCase):
```

Comment:
Can you use `parameterized` decorator for parameterized test cases? Example usage is [here](https://github.com/pfnet/chainer/blob/a8de5718b6197ce4882eed058e467528db2e7698/tests/cupy_tests/testing_tests/test_parameterized.py)